fix: Rehaul console object (#1675)
appsmithorg/appsmith-docs: website/docs/reference/appsmith-framework/console-object.md

---
sidebar_position: 2
description: The console object provides a way to log messages to the logs tab.
---

# console

> The `console` object provides a way to log messages to the Logs tab, aiding in code debugging and monitoring. This page gives comprehensive information about the console object and its methods.

Reviewer: ```suggestion
The `console` object provides a way to output messages for code debugging. This page gives information about the `console` object and its methods.
```

## Methods

> The methods log the messages to the logs tab. The logged messages are only accessible during the ongoing session. 

Reviewer: ```suggestion
The methods log the messages in the **Logs** tab of the Javascript editor. The messages are not saved and are only available for the current session. 
```

Author: The Logs tab is available for database queries, JavaScript editor, and also for API queries.

Author: updated to add this, and modified the image.

> <figure>

Reviewer: let's move the image outside and just above the **Methods** section.

  <img src="/img/global-objects-console-logs-tab.png" style= {{width:"700px", height:"auto"}} alt="The Logs tab"/>
  <figcaption align = "center"><i>Logged messages</i></figcaption>
</figure>

> In Appsmith, you can use the following methods for logging messages:

Reviewer: ```suggestion
```

:::note

> Console methods do not support string substitutions.

Reviewer: keep this as regular text outside of note callout

:::

> * [log](#consolelog-args)

Reviewer: delete all these bullet points. We are not following this structure anywhere for methods
